HealthVault/Classes/Store/MHVKeychainServiceProtocol.h

```diff
@@ -66,6 +66,13 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (BOOL)removeObjectForKey:(NSString *)key;
 
+/**
+ Generate a very long random password, suitable for using as Cache database encryption key
+ 
+ @return The random string.
+ */
+- (NSString *)generateRandomPassword;
```

Comment:
This doesn't belong in the keychain service. You should create a new class that does this or just add it as a method to the MHVThingCacheDatabase.